datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/MonthDeserializer.java

```diff
@@ -0,0 +1,48 @@
+package com.fasterxml.jackson.datatype.jsr310.deser;
+
+import java.io.IOException;
+import java.time.Month;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.deser.std.DelegatingDeserializer;
+import com.fasterxml.jackson.databind.exc.InvalidFormatException;
+
+public class MonthDeserializer extends DelegatingDeserializer {
+    private final boolean _enableOneBaseMonths;
+
+    public MonthDeserializer(JsonDeserializer<Enum> defaultDeserializer, boolean oneBaseMonths) {
+        super(defaultDeserializer);
+        _enableOneBaseMonths = oneBaseMonths;
+    }
+
+    @Override
+    public Object deserialize(JsonParser parser, DeserializationContext context) throws IOException {
+        Month zeroBaseMonth = (Month) getDelegatee().deserialize(parser, context);
+        if (!_enableOneBaseMonths || zeroBaseMonth == null) {
```

Comment:
I think I would actually change this so that if `! _enableOneBaseMonths`, or current token not `JsonToken.VALUE_NUMBER_INT` (and I guess "Stringified" number),  just call `delegate.deserialize()` and return value you get.

This is based on understanding that, I think, String name like `"FEBRUARY"` should NOT be changed to `JANUARY`? At least to me 0- vs 1-based logic only applies to numeric (and stringified numbers) case, not to textual.